app/scripts/lib/setupSentry.test.js

```diff
@@ -0,0 +1,241 @@
+        request: {},
+      };
+      const rewrittenReport = rewriteReport(testReport);
+      expect(rewrittenReport.message).toStrictEqual(
+        'This 0x** address used ** on Saturday',
+      );
+    });
+
+    it('should not modify an error message with no urls or addresses', () => {
+      const testReport = {
+        message: 'This is a simple report',
+        request: {},
+      };
+      const rewrittenReport = rewriteReport(testReport);
+      expect(rewrittenReport.message).toStrictEqual('This is a simple report');
+    });
+  });
+
+  describe('removeUrlsFromBreadCrumb', () => {
+    it('should hide the breadcrumb data url', () => {
+      const testBreadcrumb = {
+        data: {
+          url: 'https://example.com',
+        },
+      };
+      const rewrittenBreadcrumb = removeUrlsFromBreadCrumb(testBreadcrumb);
+      expect(rewrittenBreadcrumb.data.url).toStrictEqual('');
+    });
+
+    it('should hide the breadcrumb data to page', () => {
+      const testBreadcrumb = {
+        data: {
+          to: 'https://example.com',
+        },
+      };
+      const rewrittenBreadcrumb = removeUrlsFromBreadCrumb(testBreadcrumb);
+      expect(rewrittenBreadcrumb.data.to).toStrictEqual('');
+    });
+
+    it('should hide the breadcrumb data from page', () => {
+      const testBreadcrumb = {
+        data: {
+          from: 'https://example.com',
+        },
+      };
+      const rewrittenBreadcrumb = removeUrlsFromBreadCrumb(testBreadcrumb);
+      expect(rewrittenBreadcrumb.data.from).toStrictEqual('');
+    });
+
+    it('should NOT hide the breadcrumb data url if the url is on the extension protocol', () => {
+      const testBreadcrumb = {
+        data: {
+          url: 'chrome-extension://abcefg/home.html',
+        },
+      };
+      const rewrittenBreadcrumb = removeUrlsFromBreadCrumb(testBreadcrumb);
+      expect(rewrittenBreadcrumb.data.url).toStrictEqual(
+        'chrome-extension://abcefg/home.html',
+      );
+    });
+
+    it('should NOT hide the breadcrumb data to page if the url is on the extension protocol', () => {
+      const testBreadcrumb = {
+        data: {
+          to: 'chrome-extension://abcefg/home.html',
+        },
+      };
+      const rewrittenBreadcrumb = removeUrlsFromBreadCrumb(testBreadcrumb);
+      expect(rewrittenBreadcrumb.data.to).toStrictEqual(
+        'chrome-extension://abcefg/home.html',
+      );
+    });
+
+    it('should NOT hide the breadcrumb data from page if the url is on the extension protocol', () => {
+      const testBreadcrumb = {
+        data: {
+          from: 'chrome-extension://abcefg/home.html',
+        },
+      };
+      const rewrittenBreadcrumb = removeUrlsFromBreadCrumb(testBreadcrumb);
+      expect(rewrittenBreadcrumb.data.from).toStrictEqual(
+        'chrome-extension://abcefg/home.html',
+      );
+    });
+
+    it('should hide to but not from or url if to is the only one not matching an internal url', () => {
```

Comment:
Nit: could we quote "to" and "from" in these descriptions? :sweat_smile: It took me *way* too long to parse this description.